### PR TITLE
Update to latest quincy code

### PR DESCRIPTION
**Pull request by andriytk**
Merge with the latest Quincy branch (17.2.4, up to commit 85f6f426633).

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -394,8 +406,11 @@ option(WITH_RADOSGW_AMQP_ENDPOINT "Rados Gateway's pubsub support for AMQP push
 option(WITH_RADOSGW_KAFKA_ENDPOINT "Rados Gateway's pubsub support for Kafka push endpoint" ON)
 option(WITH_RADOSGW_LUA_PACKAGES "Rados Gateway's support for dynamically adding lua packagess" ON)
 option(WITH_RADOSGW_DBSTORE "DBStore backend for Rados Gateway" ON)
-option(WITH_RADOSGW_MOTR "CORTX-Motr backend for Rados Gateway" ON)
-option(WITH_RADOSGW_SELECT_PARQUET "Support for s3 select on parquet objects" OFF)
+option(WITH_RADOSGW_MOTR "CORTX-Motr backend for Rados Gateway" OFF)
```
Comment (andriytk) [inline CMakeLists.txt:409]: @shailesh-vaidya, FYI - it will not build with Motr by default any more.

**Review by sachinpunadikar — APPROVED** [review]
LGTM

File: CMakeLists.txt (hunk 1 above)
Comment (andriytk) [reply CMakeLists.txt:409]: In the confluence page which I created it's already mentioned that `-DWITH_RADOSGW_MOTR=YES` should be used. I don't know about other pages or scripts. But this change was requested by @shailesh-vaidya. I suppose, he is responsible for the build scripts, no? If not or not only him - please let others know. Thanks.

**Comment by andriytk** [comment]
retest this please

**Comment by andriytk** [comment]
retest this please

**Comment by shailesh-vaidya** [comment]
retest this please

**Comment by andriytk** [comment]
retest this please

**Pull request merged**